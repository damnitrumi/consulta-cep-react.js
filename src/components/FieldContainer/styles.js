import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    gap: ${theme.spacings.tiny};
    margin-bottom: ${theme.spacings.medium};
  `}
`;
