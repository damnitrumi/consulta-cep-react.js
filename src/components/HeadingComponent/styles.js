import styled, { css } from "styled-components";

export const Container = styled.h1`
  ${({ theme }) => css`
    max-width: auto;
    margin: ${theme.spacings.tiny} 0;
    color: red;
  `}
`;
