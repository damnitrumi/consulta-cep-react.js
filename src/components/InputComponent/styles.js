import styled, { css } from "styled-components";

export const Container = styled.input`
  ${({ theme }) => css`
    padding: ${theme.spacings.tiny};
    outline-color: #0fccff;
    font-size: ${theme.font.sizes.small};
  `}
`;
