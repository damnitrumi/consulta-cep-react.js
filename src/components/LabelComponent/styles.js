import styled, { css } from "styled-components";

export const Container = styled.label`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
  `}
`;
