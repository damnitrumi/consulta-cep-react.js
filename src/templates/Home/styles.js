import styled, { css } from "styled-components";

export const Container = styled.div`
  ${({ theme }) => css`
    padding: ${theme.spacings.small};
  `}
`;
