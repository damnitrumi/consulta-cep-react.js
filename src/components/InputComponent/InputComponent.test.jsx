import { screen } from "@testing-library/react";
import { renderTheme } from "../../styles/render-theme";
import { InputComponent } from ".";
import userEvent from "@testing-library/user-event";

describe("<InputComponent />", () => {
  it("should render with function and call function", () => {
    const fn = jest.fn();
    renderTheme(<InputComponent onBlur={fn} id="teste" />);

    const input = screen.getByPlaceholderText(/Digite seu CEP/i);
    userEvent.type(input, "00000000");
    userEvent.tab();
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it("should render with function and call function snapshot", () => {
    const fn = jest.fn();
    renderTheme(<InputComponent onBlur={fn} id="teste" />);

    const input = screen.getByPlaceholderText(/Digite seu CEP/i);
    expect(input).toMatchSnapshot();
  });

  it("should render without function", () => {
    renderTheme(<InputComponent id="teste" />);
    const input = screen.getByPlaceholderText(
      "Digite seu CEP no primeiro campo"
    );

    expect(input).toHaveAttribute(
      "placeholder",
      "Digite seu CEP no primeiro campo"
    );
  });

  it("should render without function snapshot", () => {
    renderTheme(<InputComponent id="teste" />);
    const input = screen.getByPlaceholderText(
      "Digite seu CEP no primeiro campo"
    );

    expect(input).toMatchSnapshot();
  });
});
