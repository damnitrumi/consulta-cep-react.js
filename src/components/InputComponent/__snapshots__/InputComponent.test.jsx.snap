// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InputComponent /> should render with function and call function snapshot 1`] = `
.c0 {
  padding: 0.8rem;
  outline-color: #0fccff;
  font-size: 1.6rem;
}

<input
  class="c0"
  id="teste"
  placeholder="Digite seu CEP (ex: 00000-000) e pressione a tecla 'tab' ou clique fora do campo de pesquisa"
  type="text"
/>
`;

exports[`<InputComponent /> should render without function snapshot 1`] = `
.c0 {
  padding: 0.8rem;
  outline-color: #0fccff;
  font-size: 1.6rem;
}

<input
  class="c0"
  id="teste"
  placeholder="Digite seu CEP no primeiro campo"
  readonly=""
  type="text"
  value=""
/>
`;
